src/test/java/bio/terra/janitor/integration/TrackResourceIntegrationTest.java

```diff
@@ -85,22 +84,26 @@ public void subscribeTrackResource() throws Exception {
     publisher.publish(PubsubMessage.newBuilder().setData(data).build());
     Thread.sleep(5000);
 
-    TrackedResource trackedResourceInfo = janitorDao.retrieveTrackedResource(RESOURCE_UID).get();
+    String getResponse =
+        this.mvc
+            .perform(
+                get("/api/janitor/v1/resource")
+                    .queryParam("cloudResourceUid", objectMapper.writeValueAsString(RESOURCE_UID)))
+            .andDo(MockMvcResultHandlers.print())
+            .andExpect(status().isOk())
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
 
-    assertEquals(RESOURCE_UID, trackedResourceInfo.cloudResourceUid());
+    TrackedResourceInfoList resourceInfoList =
+        objectMapper.readValue(getResponse, TrackedResourceInfoList.class);
+    assertEquals(1, resourceInfoList.getResources().size());
+    TrackedResourceInfo trackedResourceInfo = resourceInfoList.getResources().get(0);
+    assertEquals(RESOURCE_UID, trackedResourceInfo.getResourceUid());
     assertEquals(
         Duration.ofMinutes(TIME_TO_LIVE_MINUTE),
-        Duration.between(trackedResourceInfo.creation(), trackedResourceInfo.expiration()));
-    assertEquals(TrackedResourceState.READY, trackedResourceInfo.trackedResourceState());
-  }
-
-  private static ServiceAccountCredentials getGoogleCredentialsOrDie(String serviceAccountPath) {
-    try {
-      return ServiceAccountCredentials.fromStream(
-          Thread.currentThread().getContextClassLoader().getResourceAsStream(serviceAccountPath));
-    } catch (Exception e) {
-      throw new RuntimeException(
-          "Unable to load GoogleCredentials from " + serviceAccountPath + "\n", e);
-    }
+        Duration.between(trackedResourceInfo.getCreation(), trackedResourceInfo.getExpiration()));
```

Comment:
So now that we're using pub/sub, I don't think we should rely on the time we receive the message to account for the creation time & set the ttl. by the nature of pub/sub, that could be a long time after the resource was created. That was fine for the post endpoint, but I think we should change the format of the CreateResourceRequestBody to include the creation and expiration time now. Fine to do in a follow up PR, or now. this will be a breaking change to our api, so we'll have to fix up the CRL usage, but we have to change it anyway for pub/sub.

Reply:
I think we can get publish time from message. Can we use that? Changed to use that timestamp for now, and will have some follow up PRs if we want client to set the time. But I feel the publish time is fine to use?

Reply:
Having an explicit creation time parameter seems neater to me. Also makes our tests a little easier to write.

Reply:
Done